src/analysis/functions.rs

```diff
@@ -192,9 +194,13 @@ fn analyze_function(
         if let Some((callback_type, bound_name)) = type_string {
             let func_name = func.c_identifier.as_ref().unwrap();
             let finish_func_name = replace_async_by_finish(func_name);
-            let rust_finish_func_name = replace_async_by_finish(&func.name);
             let mut output_params = vec![];
-            if let Some(function) = find_function(env, &rust_finish_func_name) {
+            let mut ffi_ret = None;
+            if let Some(function) = find_function(env, &finish_func_name) {
```

Comment:
We still add trampoline when no finish function found, its normal?
If not then `output_params` and `ffi_ret` can be moved inside this if and created without `mut` modifier.

Reply:
Well I don't really know if that's normal, that's not really related to this PR though is it?

Reply:
Yes